v3/src/models/document/free-tile-row.ts

```diff
@@ -56,7 +57,7 @@ export interface IFreeTileInRowOptions extends ITileInRowOptions {
   height?: number
 }
 export const isFreeTileInRowOptions = (options?: ITileInRowOptions): options is IFreeTileInRowOptions =>
-              (options as any)?.x != null && (options as any)?.y != null
+              (options as any)?.x != null && (options as any).y != null
```

Comment:
Could you drop the `?` from `(options as any)?.x` like you dropped it from the check for `y`? And could you use `IFreeTileInRowOptions` in place of `any`?

Reply:
I tweaked the test to eliminate the cast altogether.